demo

```diff
@@ -144,84 +144,97 @@ MENDER_SERVER_URI="https://localhost"
 
 # Block until the useradm service returns an HTTP 4xx response
 echo "Waiting for the useradm service to start up..."
-until $USERADM_SERVICE_UP; do
-    $(curl --silent -k -X POST -u $MENDER_SERVER_USER\
-           --fail\
-           --connect-timeout 120\
-           --retry 60\
-           --retry-delay 2\
-           --retry-max-time 120\
-           $MENDER_SERVER_URI/api/management/v1/useradm/auth/login)
+while :
+do
+    curl --silent -k -X POST -u $MENDER_SERVER_USER\
+         --fail\
+         --connect-timeout 120\
+         --retry 60\
+         --retry-delay 2\
+         --retry-max-time 120\
+         $MENDER_SERVER_URI/api/management/v1/useradm/auth/login
     retval=$?
     case $retval in
-        0)  USER_ADM_SERVICE_UP=true ;; # User exists - continue.
-        22) USER_ADM_SERVICE_UP=true ;; # Server 400 error, ie, server is up - continue.
+        0)  break ;; # User exists - continue.
+        22) break ;; # Server 400 error, ie, server is up - continue.
         *) echo "Failed to access the mender-useradm service"
            echo "Please check the docker-compose setup."
            echo "curl exit code" $retval
-           exit $retval
+           # exit $retval
```

Comment:
This will retry infinitely. I think we should give up after three attempts or so. Also, I think we should change the message with the new behavior. Maybe something like "Failed to access the mender-useradm service. Retrying...", and then save the other part for when we give up.

The same goes for the other retry loops.